picky-signtool/src/verify.rs

```diff
@@ -0,0 +1,239 @@
+        .open(file_path)
+        .with_context(|| format!("Failed to open {:?} for reading", file_path))?;
+
+    let mut buffer = Vec::new();
+    file.read_to_end(&mut buffer).unwrap();
+
+    let (decoded, _) = encoding::decode(
+        &buffer,
+        DecoderTrap::Strict,
+        encoding::all::UTF_8 as encoding::EncodingRef,
+    );
+
+    let buffer = decoded.map_err(|err| anyhow!("Failed to decoded {:?} ps file: {}", file_path, err))?;
+
+    let signature = extract_ps_authenticode_signature(buffer)
+        .with_context(|| format!("Failed to extract Authenticode signature from {:?}", file_path))?;
+
+    let der_signature = base64::decode(signature).context("Failed to convert signature to DER")?;
+
+    let authenticode_signature = AuthenticodeSignature::from_der(&der_signature)
+        .with_context(|| format!("Failed to deserialize Authenticode signature for {:?}", file_path))?;
+
+    Ok(authenticode_signature)
+}
+
+fn extract_ps_authenticode_signature(content: String) -> anyhow::Result<String> {
+    let index = content
+        .find(PS_AUTHENTICODE_HEADER)
+        .ok_or_else(|| anyhow!("File is not digital signed"))?;
+
+    let (_, signature) = content.split_at(index);
+    let mut out = String::new();
+    for line in signature.lines() {
+        if line.contains(PS_AUTHENTICODE_HEADER) || line.contains(PS_AUTHENTICODE_FOOTER) {
+            continue;
+        }
+
+        if line.contains(PS_AUTHENTICODE_LINES_SPLITTER) {
+            out.push_str(line.replace(PS_AUTHENTICODE_LINES_SPLITTER, "").trim_end_matches('\r'));
+        }
+    }
+
+    Ok(out)
+}
+
+fn apply_flags<'a>(
+    validator: &'a AuthenticodeValidator<'a>,
+    flags: &[String],
+    time: &'a UTCDate,
+    file_hash: Vec<u8>,
+) -> &'a AuthenticodeValidator<'a> {
+    let validator = validator
+        .ignore_basic_authenticode_validation()
+        .ignore_signing_certificate_check()
+        .ignore_chain_check()
+        .ignore_ca_against_ctl_check()
+        .ignore_not_before_check()
+        .ignore_not_after_check()
+        .ignore_excluded_cert_authorities();
+
+    let validator = if flags.iter().any(|flag| flag.as_str() == ARG_VERIFY_DEFAULT) {
+        validator.require_basic_authenticode_validation(file_hash)
+    } else {
+        &validator
+    };
+
+    let validator = if flags.iter().any(|flag| flag.as_str() == ARG_VERIFY_SIGNING_CERTIFICATE) {
+        validator
+            .require_signing_certificate_check()
+            .require_not_after_check()
+            .require_not_before_check()
+            .require_chain_check()
+            .exact_date(time)
+    } else {
+        &validator
+    };
+
+    let validator = if flags.iter().any(|flag| flag.as_str() == ARG_VERIFY_CHAIN) {
+        validator.require_chain_check()
+    } else {
+        &validator
+    };
+
+    if flags.iter().any(|flag| flag.as_str() == ARG_VERIFY_CA) {
+        validator.require_ca_against_ctl_check()
+    } else {
+        validator
+    }
+}
```

Comment:
"DEFAULT" flag is probably a bit too lenient. Do you agree with this approach? @awakecoding

Reply:
Basic sounds good to me, we can always add "extended" or something elsewhere after